lib/spack/spack/ci.py

```diff
@@ -40,6 +42,8 @@
 spack_gpg = SpackCommand('gpg')
 spack_compiler = SpackCommand('compiler')
 
+runner_var_regex = re.compile('\\$env:(.+)$')
```

Comment:
Would this regex catch something like this?

`my_var: EXTRA_GRABAGE$env:MY_VAR`

Is this desired, or should this match only on values that also *start* with `$env`?

If the latter, consider adding `^` to the regex (i.e.: `'^\\$env:(.+)$'`)

Reply:
Hmmm, seems like allowing it to replace a substring, as you showed in your extra garbage example, could be useful.  In which case I don't want to bind to the beginning of the string, but I would need to make sure to replace only the matched portion with the interpolated variable, rather than the whole string.

I'll add this change, but keep it as a separate commit.  If people have strong objections, I can always remove it.